src/osyris/spatial/spatial.py

```diff
@@ -0,0 +1,43 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2022 Osyris contributors (https://github.com/nvaytet/osyris)
+
+import numpy as np
+
+
+def extract_sphere(dataset, radius, origin):
+    """
+    Extract a spherical subdomain around an origin point.
+    """
+    subdomain = dataset.__class__(nout=dataset.meta["nout"], path=dataset.meta["path"])
```

Comment:
We need to think a little more about what's going on here.

This comes from the fact that the Dataset and the Loader are entangled in the same object. When creating a new Dataset, we make a new loader, which is fine
But then we load the metadata again from disk here https://github.com/osyris-project/osyris/blob/main/src/osyris/core/dataset.py#L20

Maybe we want to avoid that? (I think we are doing he same in the `copy` method of the Dataset, so we would need to change that as well to be consistent)

Reply:
> Perhaps we could add a method in Dataset that achieves this?

Sorry I didn't really get what "this" is refering to here. I would have thought a shallow copy would be ok. You can add new entries, delete old entries. I'm not sure how often we are modifying values in the meta. Mostly just reading from it?

Reply:
I just checked, and shallow copies can indeed work if you slice the group (which I forgot is possible, still mindblown btw). I'll still have to check if the given group has data in the subdomain, so that we don't have groups that are references to the old dataset in the new one.